docs/docs/reference/contextual/derivation.md

```diff
@@ -249,80 +237,70 @@ import scala.deriving._
 import scala.compiletime.{erasedValue, summonInline}
 
 inline def summonAll[T <: Tuple]: List[Eq[_]] =
-  inline erasedValue[T] match {
-    case _: EmptyTuple => Nil
-    case _: (t *: ts) => summonInline[Eq[t]] :: summonAll[ts]
-  }
-
-trait Eq[T] {
-  def eqv(x: T, y: T): Boolean
-}
-
-object Eq {
-  
-  given Eq[Int] with {
-    def eqv(x: Int, y: Int) = x == y
-  }
-
-  def check(elem: Eq[_])(x: Any, y: Any): Boolean =
-    elem.asInstanceOf[Eq[Any]].eqv(x, y)
-
-  def iterator[T](p: T) = p.asInstanceOf[Product].productIterator
-
-  def eqSum[T](s: Mirror.SumOf[T], elems: => List[Eq[_]]): Eq[T] =
-    new Eq[T] {
-      def eqv(x: T, y: T): Boolean = {
-        val ordx = s.ordinal(x)
-        (s.ordinal(y) == ordx) && check(elems(ordx))(x, y)
-      }
-    }
-
-  def eqProduct[T](p: Mirror.ProductOf[T], elems: => List[Eq[_]]): Eq[T] =
-    new Eq[T] {
-      def eqv(x: T, y: T): Boolean =
-        iterator(x).zip(iterator(y)).zip(elems.iterator).forall {
-          case ((x, y), elem) => check(elem)(x, y)
-        }
-    }
-
-  inline given derived[T](using m: Mirror.Of[T]): Eq[T] = {
-    lazy val elemInstances = summonAll[m.MirroredElemTypes]
-    inline m match {
+   inline erasedValue[T] match
+   case _: EmptyTuple => Nil
+   case _: (t *: ts) => summonInline[Eq[t]] :: summonAll[ts]
+
+trait Eq[T]:
+   def eqv(x: T, y: T): Boolean
+
+object Eq:
+   given Eq[Int] with
+      def eqv(x: Int, y: Int) = x == y
+
+   def check(elem: Eq[_])(x: Any, y: Any): Boolean =
+      elem.asInstanceOf[Eq[Any]].eqv(x, y)
+
+   def iterator[T](p: T) = p.asInstanceOf[Product].productIterator
+
+   def eqSum[T](s: Mirror.SumOf[T], elems: => List[Eq[_]]): Eq[T] =
+      new Eq[T]:
+         def eqv(x: T, y: T): Boolean =
```

Comment:
That was an error. I used `with` first and then added a commit to change to `:` which is the currently supported syntax (and it might stay that way, we have not decided yet). But the `new Vehicle` line escaped by regex search since the `with` was followed by a comment.